lua/InternalEvents.cpp

```diff
@@ -31,6 +31,108 @@ HOOK_METHOD(CApp, OnKeyDown, (SDLKey key) -> void)
     if (!preempt) super(key);
 }
 
+HOOK_METHOD(CApp, OnKeyUp, (SDLKey key) -> void)
+{
+    LOG_HOOK("HOOK_METHOD -> CApp::OnKeyUp -> Begin (InternalEvents.cpp)\n")
+
+    auto context = Global::GetInstance()->getLuaContext();
+
+    lua_pushinteger(context->GetLua(), key);
+    bool preempt = context->getLibScript()->call_on_internal_chain_event_callbacks(InternalEvents::ON_KEY_UP, 1, 0);
+    lua_pop(context->GetLua(), 1);
+
+    if (!preempt) super(key);
+}
+
+HOOK_METHOD(CApp, OnMouseMove, (int x, int y, int xdiff, int ydiff, bool holdingLMB, bool holdingRMB, bool holdingMMB) -> void)
+{
+    LOG_HOOK("HOOK_METHOD -> CApp::OnMouseMove -> Begin (InternalEvents.cpp)\n")
+
+    auto context = Global::GetInstance()->getLuaContext();
+
+    lua_pushinteger(context->GetLua(), x);
+    lua_pushinteger(context->GetLua(), y);
+    lua_pushinteger(context->GetLua(), xdiff);
+    lua_pushinteger(context->GetLua(), ydiff);
+    lua_pushboolean(context->GetLua(), holdingLMB);
+    lua_pushboolean(context->GetLua(), holdingRMB);
+    lua_pushboolean(context->GetLua(), holdingMMB);
+    bool preempt = context->getLibScript()->call_on_internal_chain_event_callbacks(InternalEvents::ON_MOUSE_MOVE, 7, 0);
+    lua_pop(context->GetLua(), 7);
+
+    if (!preempt) super(x, y, xdiff, ydiff, holdingLMB, holdingRMB, holdingMMB);
+}
+
+HOOK_METHOD(CApp, OnLButtonDown, (int x, int y) -> void)
+{
+    LOG_HOOK("HOOK_METHOD -> CApp::OnLButtonDown -> Begin (InternalEvents.cpp)\n")
+
+    auto context = Global::GetInstance()->getLuaContext();
+
+    lua_pushinteger(context->GetLua(), x);
+    lua_pushinteger(context->GetLua(), y);
+    bool preempt = context->getLibScript()->call_on_internal_chain_event_callbacks(InternalEvents::ON_MOUSE_L_BUTTON_DOWN, 2, 0);
+    lua_pop(context->GetLua(), 2);
+
+    if (!preempt) super(x, y);
+}
+
+HOOK_METHOD(CApp, OnLButtonUp, (int x, int y) -> void)
+{
+    LOG_HOOK("HOOK_METHOD -> CApp::OnLButtonUp -> Begin (InternalEvents.cpp)\n")
+
+    auto context = Global::GetInstance()->getLuaContext();
+
+    lua_pushinteger(context->GetLua(), x);
+    lua_pushinteger(context->GetLua(), y);
+    bool preempt = context->getLibScript()->call_on_internal_chain_event_callbacks(InternalEvents::ON_MOUSE_L_BUTTON_UP, 2, 0);
+    lua_pop(context->GetLua(), 2);
+
+    if (!preempt) super(x, y);
+}
+
+HOOK_METHOD(CApp, OnRButtonDown, (int x, int y) -> void)
+{
+    LOG_HOOK("HOOK_METHOD -> CApp::OnRButtonDown -> Begin (InternalEvents.cpp)\n")
+
+    auto context = Global::GetInstance()->getLuaContext();
+
+    lua_pushinteger(context->GetLua(), x);
+    lua_pushinteger(context->GetLua(), y);
+    bool preempt = context->getLibScript()->call_on_internal_chain_event_callbacks(InternalEvents::ON_MOUSE_R_BUTTON_DOWN, 2, 0);
+    lua_pop(context->GetLua(), 2);
+
+    if (!preempt) super(x, y);
+}
+
+HOOK_METHOD_PRIORITY(CApp, OnRButtonUp, -100, (int x, int y) -> void)
```

Comment:
the other hook for `OnRButtonUp` in `CustomCommandGui.cpp` doesn't call `super` and prevents this from working without a priority